torch/_inductor/codegen/scheduling.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+
+class Scheduling(ABC):
+    @abstractmethod
+    def can_fuse_vertical(self, node1, node2):
+        pass
+
+    @abstractmethod
+    def can_fuse_horizontal(self, node1, node2):
+        pass
+
+    @abstractmethod
+    def group_fn(self, *args, **kwargs):
+        pass
+
+    @abstractmethod
+    def codegen_template(self, *args, **kwargs):
```

Comment:
why is it not template node and epilogues?

Reply:
I don't think this will work with non-triton backends currently.